src/Core/ILoggerExtensions.cs

```diff
@@ -18,20 +18,32 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-using System.Diagnostics;
+using SonarLint.VisualStudio.Core;
 
 namespace SonarLint.VisualStudio.Integration.Helpers
 {
     public static class ILoggerExtensions
     {
+        private static bool shouldLogDebug;
+
+        static ILoggerExtensions() => Initialize(new EnvironmentSettings());
+
+        internal /* for testing*/ static void Initialize(IEnvironmentSettings settings)
+        {
+            shouldLogDebug = settings.LogDebugMessages();
+        }
+
```

Comment:
This is a bit more complicated now to make it testable. Perhaps I should just have written the property as
```
internal /* for testing */ static bool shouldLogDebug = new EnvironmentSettings().LogDebugMessages();
```